client/components/search/index.jsx

```diff
@@ -298,7 +307,7 @@ const Search = React.createClass( {
 				<input
 					type="search"
 					id={ 'search-component-' + this.state.instanceId }
-					className={ 'search__input' + ( this.props.dir ? ' ' + this.props.dir : '' ) }
+					className={ 'search__input' + isCloseButtonVisible + ( this.props.dir ? ' ' + this.props.dir : '' ) }
```

Comment:
@hoverduck Items now have classes that should allow to distinguish them. Please check if this is what you need.